File: projects/order-creation-queries/repository/src/main/java/com/robinfood/repository/orderdetail/IOrderDetailRepository.java
package com.robinfood.repository.orderdetail;

import com.robinfood.core.dtos.OrderDetailDTO;
import com.robinfood.core.enums.Result;

import java.util.List;

/**
 * Repository for order detail related data
 */
public interface IOrderDetailRepository {

    /**
     * Retrieves the order id's based on the following query params
     * @param orderIds the list of orders by id
     * @param orderUids the list of orders by uid
     * @return the order detail containing the orders detailed info
     */
    Result<List<OrderDetailDTO>> getOrderDetail(
            String token,
            List<Long> orderIds,
            List<String> orderUids,
            List<String> orderUuid
    );

}

File: projects/posv2/repository/src/main/kotlin/com/robinfood/repository/orderdetail/IOrderDetailRepository.kt
package com.robinfood.repository.orderdetail

import com.robinfood.core.dtos.OrderDetailDTO
import com.robinfood.core.enums.Result

/**
 * Repository for order detail related data
 */
interface IOrderDetailRepository {

    /**
     * Retrieves the orders detail based on orders identifiers given
     * [token] the authentication token to be used
     * [countryId] the country id
     * [flowId] the flow id
     * [orderIds] the orders identifiers
     * @return the orders detailed info
     */
    suspend fun getOrdersDetail(
        token: String,
        countryId: Long,
        flowId: Long,
        orderIds: List<Long>
    ): Result<List<OrderDetailDTO>>
}